ansible_ai_connect_chatbot/src/App.test.tsx

```diff
@@ -175,6 +183,38 @@ describe("App tests", () => {
 
     const sureButton = screen.getByText("Sure!");
     await act(async () => fireEvent.click(sureButton));
+
+    expect(ghIssueLinkSpy.mock.calls.length).toEqual(1);
+    expect(ghIssueLinkSpy.mock.calls[0][0]?.toString().length).toEqual(477);
+    expect(ghIssueLinkSpy.mock.calls[0][0]).toContain(
+      encodeURIComponent(referencedDocumentExample),
+    );
+  });
+
+  it("Too many reference documents for the GU issue creation query param.", async () => {
+    const ghIssueLinkSpy = vi.spyOn(global, "open");
+    mockAxios(200, false, false, referencedDocumentExample.repeat(15));
+    renderApp();
+    const textArea = screen.getByLabelText("Send a message...");
+    await act(async () => userEvent.type(textArea, "Hello"));
+    const sendButton = screen.getByLabelText("Send button");
+    await act(async () => fireEvent.click(sendButton));
+    expect(
+      screen.getByText(
+        "In Ansible, the precedence of variables is determined by the order...",
+      ),
+    ).toBeInTheDocument();
+    expect(screen.getByText("Create variables")).toBeInTheDocument();
+
+    const thumbsDownIcon = screen.getByRole("button", { name: "Bad response" });
+    await act(async () => fireEvent.click(thumbsDownIcon));
+
+    const sureButton = screen.getByText("Sure!");
+    await act(async () => fireEvent.click(sureButton));
+
+    expect(ghIssueLinkSpy.mock.calls.length).toEqual(1);
+    // Assert the size of the URL does not exceeds max length for 30 docs.
+    expect(ghIssueLinkSpy.mock.calls[0][0]?.toString().length).toEqual(2885);
```

Comment:
Another comment explaining _how_ you calculated `2885` would be nice...

...or, if it was a calculation that can be written in code; please use it.

Reply:
good catch, it was tricky... I have improved the test cases, please take a look at let me know if they're easier to follow. Thx